src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,71 @@
-import React from 'react';
+import cn from 'classnames';
+import React, {
+  ChangeEvent,
+  FormEvent,
+  useCallback,
+  useState,
+} from 'react';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { normalizeMovie } from '../../helpers/normalizeMovie';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+interface Props {
+  onAddMovie: (movie: Movie) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({ onAddMovie }) => {
+  const [query, setQuery] = useState('');
+  const [previewMovie, setPreviewMovie] = useState<Movie | null>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [onloadError, setOnloadError] = useState(false);
+
+  const handleSearchChange = useCallback((
+    event: ChangeEvent<HTMLInputElement>,
+  ) => {
+    setOnloadError(false);
+    setQuery(event.target.value);
+  }, []);
+
+  const handleFormSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setIsLoading(true);
+
+    const fetchData = async () => {
+      try {
+        const data = await getMovie(query.trim());
+
+        if ('Error' in data) {
+          setOnloadError(true);
+        } else {
+          const preparedMovieData = normalizeMovie(data);
+
+          setPreviewMovie(preparedMovieData);
+        }
+      } catch (error) {
+        setOnloadError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchData();
+  };
```

Comment:
Do you need this `fetchData` function inside? 

```suggestion
  const handleFormSubmit = async (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    setIsLoading(true);
    
    try {
      const data = await getMovie(query.trim());

      if ('Error' in data) {
        setOnloadError(true);
      } else {
        const preparedMovieData = normalizeMovie(data);

        setPreviewMovie(preparedMovieData);
      }
    } catch (error) {
      setOnloadError(true);
    } finally {
      setIsLoading(false);
    }
  };
```